bindings/nodejs/src/lib.rs

```diff
@@ -30,6 +29,11 @@ use futures::AsyncReadExt;
 use futures::TryStreamExt;
 use napi::bindgen_prelude::*;
 
+use opendal::operator_functions::{FunctionWrite, FunctionWriter};
```

Comment:
It's better not to play with those APIs. What do you think about adding bunch of `if-else` directly? 

---

This PR also makes me thinking about exporting `WriteOptions` from the rust core side. We can have a discussion there.

Reply:
you mean this?
```diff
         let mut writer = self.0.write_with(&path, c);
         if let Some(options) = options {
-            writer = writer.with(options);
+            if let Some(append) = options.append {
+                writer = writer.append(append);
+            }
+            if let Some(chunk) = options.chunk {
+                writer = writer.chunk(chunk.get_u64().1 as usize);
+            }
+            if let Some(ref content_type) = options.content_type {
+                writer = writer.content_type(content_type);
+            }
+            if let Some(ref content_disposition) = options.content_disposition {
+                writer = writer.content_disposition(content_disposition);
+            }
+            if let Some(ref cache_control) = options.cache_control {
+                writer = writer.cache_control(cache_control);
+            }
+            writer
         }
         writer.await.map_err(format_napi_error)
     }

``

Reply:
We can polish here in the future while rust core exposes `WriteOptions` directly.